### PR TITLE
fix(curriculum): add test cases to ferris-wheel project #45099

### LINKED ISSUE #45099: Correct answer marked as a bad one
## Affected page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-css-animation-by-building-a-ferris-wheel/step-14

## Your code

```
.cabin:first-of-type {
  right: -8.5%;
  top: 50%;
}

```

## Screenshots

image

## System

- Device: [e.g. iPhone6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.4]
- Browser: [e.g. chrome, safari]
- Version: [e.g. 22]

## Additional context

I could be able to solve the minichallenge with my code using the .first-of-type selector, but the minichllenge required the use of :nth-of-type(1), I think that this is ok because of the future use of the other elements, but there should be some kind of explanation saying something like "You should have a .cabin:nth-of-type(1) selector. because we will use others elements in the future using this selector: or something like this.

### PR DESCRIPTION
**Pull request by wbenarto**
Checklist:

- [ X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [ X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ X] My pull request targets the `main` branch of freeCodeCamp.
- [ X] I have tested these changes either locally on my machine, or GitPod.

Closes #45099

Step-014.md:

1. Add a test that passes .cabin:first-of-type selector
2. Add ".cabin:first-of-type" selector to the hints
3. Have the tests passes for both first-of-type and nth-of-type(1)

Step-015md:

1. Add a test that passes .cabin:last-of-type selector for the 6th element
2. Add ".cabin:last-of-type" selector to the hints
3. Have the tests passes for both last-of-type and nth-of-type(6)

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-014.md [hunk 1]
```diff
@@ -11,22 +11,22 @@ Time to position the cabins around the wheel. Select the first `.cabin` element.
 
 # --hints--
 
-You should have a `.cabin:nth-of-type(1)` selector.
+You should have a `.cabin:nth-of-type(1) or .cabin:first-of-type` selector.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-014.md:14]: I don't think we specifically need to mention the selector in the steps.

The challenge should still ask for `nth-of-type`, but allow `first-of-type` to pass.

Comment (wbenarto) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-014.md:14]: Sounds good, will push another edit to remove the first-of-type selector from hints.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-014.md [hunk 2]
```diff
@@ -14,19 +14,19 @@ Time to position the cabins around the wheel. Select the first `.cabin` element.
 You should have a `.cabin:nth-of-type(1)` selector.
 
 ```js
-assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(1)'));
+assert(new __helpers.CSSHelp(document).getStyle('.cabin:first-of-type') || new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(1)'));
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-014.md:17]:
```suggestion
const def = (s) => new __helpers.CSSHelp(document).getStyle(s);
assert.exists(def('.cabin:first-of-type') || def('.cabin:nth-of-type(1)'));
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-014.md [hunk 3]
```diff
@@ -14,19 +14,19 @@ Time to position the cabins around the wheel. Select the first `.cabin` element.
...
 ```
 
 Your `.cabin:nth-of-type(1)` selector should have a `right` property set to `-8.5%`.
 
 ```js
-assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(1)')?.right === '-8.5%');
+assert((new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(1)')?.right === '-8.5%') || (new __helpers.CSSHelp(document).getStyle('.cabin:first-of-type')?.right === '-8.5%'));
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-014.md:23]:
```suggestion
const right = (s) => new __helpers.CSSHelp(document).getStyle(s)?.right;
assert.equal(right('.cabin:first-of-type') || right('.cabin:nth-of-type(1)'),  '-8.5%'));
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-014.md [hunk 4]
```diff
@@ -14,19 +14,19 @@ Time to position the cabins around the wheel. Select the first `.cabin` element.
...
 ```
 
 Your `.cabin:nth-of-type(1)` selector should have a `top` property set to `50%`.
 
 ```js
-assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(1)')?.top === '50%');
+assert((new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(1)')?.top === '50%') || new __helpers.CSSHelp(document).getStyle('.cabin:first-of-type')?.top === '50%');
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-014.md:29]:
```suggestion
const top = (s) => new __helpers.CSSHelp(document).getStyle(s)?.top;
assert.equal(top('.cabin:first-of-type') || top('.cabin:nth-of-type(1)'),  '50%'));
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-015.md [hunk 5]
```diff
@@ -92,19 +92,19 @@ assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(5)')?.top ==
 You should have a `.cabin:nth-of-type(6)` selector.
 
 ```js
-assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(6)'));
+assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(6)') || new __helpers.CSSHelp(document).getStyle('.cabin:last-of-type'));
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-015.md:95]:
```suggestion
const def = (s) => new __helpers.CSSHelp(document).getStyle(s);
assert.exists(def('.cabin:last-of-type') || def('.cabin:nth-of-type(6)'));
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-015.md [hunk 6]
```diff
@@ -92,19 +92,19 @@ assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(5)')?.top ==
...
 ```
 
 Your `.cabin:nth-of-type(6)` selector should have a `right` property set to `17%`.
 
 ```js
-assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(6)')?.right === '17%');
+assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(6)')?.right === '17%' || new __helpers.CSSHelp(document).getStyle('.cabin:last-of-type')?.right === '17%');
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-015.md:101]:
```suggestion
const right = (s) => new __helpers.CSSHelp(document).getStyle(s)?.right;
assert.equal(right('.cabin:last-of-type') || right('.cabin:nth-of-type(6)'),  '17%'));
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-015.md [hunk 7]
```diff
@@ -92,19 +92,19 @@ assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(5)')?.top ==
...
 ```
 
 Your `.cabin:nth-of-type(6)` selector should have a `top` property set to `7%`.
 
 ```js
-assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(6)')?.top === '7%');
+assert(new __helpers.CSSHelp(document).getStyle('.cabin:nth-of-type(6)')?.top === '7%' || new __helpers.CSSHelp(document).getStyle('.cabin:last-of-type')?.top === '7%');
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-015.md:107]:
```suggestion
const top = (s) => new __helpers.CSSHelp(document).getStyle(s)?.top;
assert.equal(top('.cabin:last-of-type') || top('.cabin:nth-of-type(6)'),  '7%'));
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you, for taking the time to contribute.

Just some readability changes.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-015.md (hunk 7 above)
Comment (wbenarto) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-015.md]: @ShaunSHamilton I made these changes you suggested along for step-015's 6th element. Thank you for the tips!

Comment (ShaunSHamilton) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-css-animation-by-building-a-ferris-wheel/step-015.md]: For future reference, you can use the `Commit suggestion` button to accept a suggestion and commit it. Even better, when there are multiple suggestions, you can go to the `Files` tab, and use the `Add suggestion` to batch button to include all suggestions in a single commit.

**Review by ShaunSHamilton — APPROVED** [review]
Thank you, for making those changes.

This LGTM 👍

**Review by naomi-lgbt — APPROVED** [review]
Thanks for taking care of this!

**Pull request merged**